packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -265,12 +266,17 @@ export function convertSnapshotTreeToSummaryTree(
         0x19e /* "There should not be commit tree entries in snapshot" */);
 
     const builder = new SummaryTreeBuilder();
-    for (const [key, value] of Object.entries(snapshot.blobs)) {
-        // The entries in blobs are supposed to be blobPath -> blobId and blobId -> blobValue
-        // and we want to push blobPath to blobValue in tree entries.
-        if (snapshot.blobs[value] !== undefined) {
-            const decoded = fromBase64ToUtf8(snapshot.blobs[value]);
-            builder.addBlob(key, decoded);
+    for (const [path, id] of Object.entries(snapshot.blobs)) {
+        // 0.44 back-compat We still put contents in same blob for back-compat so need to add blob
+        // only for blobPath -> blobId mapping and not for blobId -> blob value contents.
+        if (snapshot.blobs[id] !== undefined) {
```

Comment:
I'm a bit confused by this check and following logic.
On more hand, presence of snapshot.blobs[id] means that one can immediately use use - no need to check and use blobsContents (that's legacy path).
On another hand, I'd expect check be reverse and check for presence of blobsContents first, and leverage it, and only then try legacy path (that should be marked with "// back-compat" comment such that we can find it later.